src/reducers/bitcoinReducer.js

```diff
@@ -0,0 +1,144 @@
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import type { BitcoinAddress, BitcoinUtxo } from 'models/Bitcoin';
+import {
+  UPDATE_BITCOIN_BALANCE,
+  SET_BITCOIN_ADDRESSES,
+} from 'constants/bitcoinConstants';
+
+export type BitcoinReducerState = {
+  data: {
+    addresses: BitcoinAddress[],
+    unspentTransactions: BitcoinUtxo[],
+  },
+};
+
+type BitcoinReducerActionPayload = {
+  addresses?: string[],
+  address?: string,
+  unspentTransactions?: BitcoinUtxo[],
+};
+
+export type BitcoinReducerAction = {
+  type: string,
+  payload?: BitcoinReducerActionPayload,
+};
+
+const initialState = {
+  data: {
+    addresses: [],
+    unspentTransactions: [],
+  },
+};
+
+const updateBalance = (
+  state: BitcoinReducerState,
+  payload: BitcoinReducerActionPayload,
+): BitcoinReducerState => {
+  const { unspentTransactions = [], address = '' } = payload;
+  const { data: { addresses, unspentTransactions: transactions } } = state;
+
+  const filteredTransactions: BitcoinUtxo[] = transactions.filter(
+    ({ address: txAddress }) => address !== txAddress,
+  );
+
+  const matchingAddress: ?BitcoinAddress = addresses.find(
+    ({ address: addr }) => addr === address,
+  );
+  if (!matchingAddress) {
+    return state;
+  }
+
+  const updatedAddress: BitcoinAddress = { address, updatedAt: Date.now() };
+  const filteredAddresses: BitcoinAddress[] = addresses.filter(
+    ({ address: addr }) => addr !== address,
+  );
+
+  return {
+    ...state,
+    data: {
+      addresses: [...filteredAddresses, updatedAddress],
+      unspentTransactions: filteredTransactions.concat(...unspentTransactions),
+    },
+  };
+};
+
+const setAddresses = (
+  state: BitcoinReducerState,
+  payload: BitcoinReducerActionPayload,
+): BitcoinReducerState => {
+  const { addresses = [], unspentTransactions = [] } = payload;
+
+  const addressesInfo: BitcoinAddress[] = addresses.map(
+    address => ({ address, updatedAt: 0 }),
+  );
+
+  return {
+    ...state,
+    data: { addresses: addressesInfo, unspentTransactions },
+  };
+};
+
+// const createAddress = (
+//   state: BitcoinReducerState,
+//   payload: BitcoinReducerActionPayload,
+// ): BitcoinReducerState => {
+//   const { address = '' } = payload;
+//   const { data: { addresses, unspentTransactions } } = state;
+//
+//   if (!address) {
+//     return { ...state, isCreatingAddress: false };
+//   }
+//
+//   return {
+//     ...state,
+//     data: {
+//       addresses: [
+//         ...addresses,
+//         { address, updatedAt: 0 },
+//       ],
+//       unspentTransactions,
+//     },
+//   };
+// };
```

Comment:
This will be needed later? If so maybe adding TODO would explain when and why

Reply:
Yeah, well.. This is gonna be needed before we release this on production, this was how I imagined it to be on this approach. The thing is that we wont have multiple addresses on the MVP, so this code wasn't going to actually be used.

We should probably _uncomment_ it so it's not ignored, I don't think it's gonna be a problem.